Bluepill-runner/Bluepill-runner/BPReportCollector.m

```diff
@@ -87,4 +88,93 @@ + (void)collectReportsFromPath:(NSString *)reportsPath
     [xmlData writeToFile:finalReportPath atomically:YES];
 }
 
++ (void)collectReportsFromPath:(NSString *)reportsPath
+                 withTestConfig:(NSDictionary *)testConfig
+                   applyXQuery:(NSString *)XQuery
+                 hideSuccesses:(BOOL)hideSuccesses
+          withTraceEventAtPath:(NSString *)finalReportPath {
+    NSFileManager *fileManager = [NSFileManager defaultManager];
+    
+    NSURL *directoryURL = [NSURL fileURLWithPath:reportsPath isDirectory:YES];
+    NSArray *keys = [NSArray arrayWithObject:NSURLIsDirectoryKey];
+    NSDirectoryEnumerator *enumerator = [fileManager
+                                         enumeratorAtURL:directoryURL
+                                         includingPropertiesForKeys:keys
+                                         options:0
+                                         errorHandler:^(NSURL *url, NSError *error) {
+                                             fprintf(stderr, "Failed to process url %s", [[url absoluteString] UTF8String]);
+                                             return YES;
+                                         }];
+    TraceEvent *traceEvent = [[TraceEvent alloc] initWithData:testConfig];
+    NSCharacterSet* notDigits = [[NSCharacterSet decimalDigitCharacterSet] invertedSet];
+    NSString *currentSim = @"-1";
```

Comment:
-1 looks weird. why not just empty or null?

Reply:
I'll change this to an empty string, I didn't want to use null because of the JSONSerializer.